main.js

```diff
@@ -77,8 +94,30 @@ function deleteIdea(e) {
     for (var i = 0; i < ideas.length; i++) {
       if (ideas[i].id === uniqueId) {
         ideas.splice(i, 1);
-      }
-    }
+      };
+    };
     displayIdeas();
-  }
+  };
+};
+
+function toggle(visibleElement, hiddenElement) {
+  visibleElement.classList.toggle("hidden");
+  hiddenElement.classList.toggle("hidden");
+};
+
+function toggleStarredIdeas() {
+  cardContainer.classList.toggle("show-starred");
+}
+
+function filterIdeas() {
+  var ideasToHide = ideas.filter(idea => !idea.title.includes(searchTextField.value) && !idea.body.includes(searchTextField.value));
+  var ideasToShow = ideas.filter(idea => idea.title.includes(searchTextField.value) || idea.body.includes(searchTextField.value));
+  for (var i = 0; i < ideasToHide.length; i++) {
+    var selector = document.getElementById(ideasToHide[i].id);
+    selector.classList.add('hidden');
+  };
+  for (var i = 0; i < ideasToShow.length; i++) {
+    var selector = document.getElementById(ideasToShow[i].id);
+    selector.classList.remove('hidden');
+  };
```

Comment:
I like the logic here. But I think theres a bit of "over engineering" going on. How could this be simplified? Could we do the same thing by only looping through the ideas array?